odoo/addons/base/models/ir_actions_report.py

```diff
@@ -37,25 +34,35 @@
 ImageFile.LOAD_TRUNCATED_IMAGES = True
 
 _logger = logging.getLogger(__name__)
+_barcode_init_lock = RLock()
+
 
 # A lock occurs when the user wants to print a report having multiple barcode while the server is
 # started in threaded-mode. The reason is that reportlab has to build a cache of the T1 fonts
 # before rendering a barcode (done in a C extension) and this part is not thread safe. We attempt
 # here to init the T1 fonts cache at the start-up of Odoo so that rendering of barcode in multiple
 # thread does not lock the server.
-_DEFAULT_BARCODE_FONT = 'Courier'
-try:
-    available = TypeFace(_DEFAULT_BARCODE_FONT).findT1File()
-    if not available:
-        substitution_font = 'NimbusMonoPS-Regular'
-        fnt = getFont(substitution_font)
-        if fnt:
-            _DEFAULT_BARCODE_FONT = substitution_font
-            fnt.ascent = 629
-            fnt.descent = -157
-    createBarcodeDrawing('Code128', value='foo', format='png', width=100, height=100, humanReadable=1, fontName=_DEFAULT_BARCODE_FONT).asString('png')
-except Exception:
-    pass
+@functools.lru_cache(1)
+def _barcode_font():
+    from reportlab.graphics.barcode import createBarcodeDrawing  # noqa: PLC0415
+    from reportlab.pdfbase.pdfmetrics import getFont, TypeFace  # noqa: PLC0415
```

Comment:
Can you add a comment about why we do it like this? It may avoid someone to copy the import without lazy loading or to clean this.